muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/announcer/rest/AnnouncerRESTService.java

```diff
@@ -402,26 +402,16 @@ private boolean canSeeAnnouncement(Announcement announcement, SchoolDataIdentifi
   private boolean hasOrganizationAccess(Announcement announcement, UserSchoolDataIdentifier userSchoolDataIdentifier) {
     Long announcementOrganizationId = announcement.getOrganizationEntityId();
     Long userOrganizationId = userSchoolDataIdentifier.getOrganization() != null ? userSchoolDataIdentifier.getOrganization().getId() : null;
-    EnvironmentRoleArchetype role = userSchoolDataIdentifier.getRole() != null ? userSchoolDataIdentifier.getRole().getArchetype() : null;
 
-    if (announcementOrganizationId != null) {
-      if (userOrganizationId != null && !Objects.equals(announcementOrganizationId, userOrganizationId)) {
-        // Both ids present but don't match
-        return false;
-      }
-      
-      if (userOrganizationId == null && role != EnvironmentRoleArchetype.ADMINISTRATOR) {
-        // User has no organization, but we give a pass for administrators
-        return false;
-      }
-    } else {
-      // Announcements without organization can only be found by administrators
-      if (role != EnvironmentRoleArchetype.ADMINISTRATOR) {
-        return false;
-      }
+    if (userSchoolDataIdentifier.hasRole(EnvironmentRoleArchetype.ADMINISTRATOR)) {
```

Comment:
This works a tiny bit differently than before so try to figure out the consequences.